### PR TITLE
Add GeoJSON support.

### PR DESCRIPTION
**Pull request by bziobrowski**
This PR adds support for GeoJSON format (see 1 & 2) via following functions:
```
ST_GeomFromGeoJson(string) -> binary
ST_GeogFromGeoJson(string) -> binary
ST_AsGeoJson(binary) -> string
```

Supported data types:
- Point
- LineString
- Polygon
- MultiPoint
- MultiLineString
- MultiPolygon
- GeometryCollection
- Feature
- FeatureCollection

Examples:

A. Convert from WKT to GeoJSON:

- convert Point geometry 
```sql
select ST_AsGeoJson(ST_GeomFromText('POINT (1 2)'))
```
returns  
```json
{ 
  "type":"Point",
  "coordinates":[1,2],
  "crs":{"type":"name","properties":{"name":"EPSG:0"}}
}
```

- convert point geography
```sql
select ST_AsGeoJson(ST_GeogFromText('POINT (1 2)'))
```    
returns 
```json
{
  "type":"Point",
  "coordinates":[1,2],
  "crs":{"type":"name","properties":{"name":"EPSG:4326"}}
}
```

- convert MultiLineString geometry
```sql
select ST_AsGeoJson(ST_GeomFromText('MULTILINESTRING ((1 1, 5 1), (2 4, 4 4))'))
```
returns 
```json
{
  "type":"MultiLineString",
  "coordinates":[[[1,1],[5,1]],[[2,4],[4,4]]],
  "crs":{"type":"name","properties":{"name":"EPSG:0"}}
}
```

- convert MultiLineString geography
```sql
select ST_AsGeoJson(ST_GeogFromText('MULTILINESTRING ((1 1, 5 1), (2 4, 4 4))')) 
```
returns 
```json
{
  "type":"MultiLineString",
  "coordinates":[[[1,1],[5,1]],[[2,4],[4,4]]],
  "crs":{"type":"name","properties":{"name":"EPSG:4326"}}
}
```

B. Convert from GeoJson to WKT:

- convert Point geometry
```sql
select ST_AsText(ST_GeomFromGeoJson('{"type":"Point","coordinates":[1,2]}')) 
``` 
returns
`POINT (1 2)`

- convert MultiLineString geometry
```sql
select ST_AsText(ST_GeomFromGeoJson('{"type":"MultiLineString","coordinates":[[[1,1],[5,1]],[[2,4],[4,4]]]}'')
```
returns 
`MULTILINESTRING ((1 1, 5 1), (2 4, 4 4))`

References:
1. https://geojson.org/
2. https://datatracker.ietf.org/doc/html/rfc7946

### CODE REVIEW

**Comment by Jackie-Jiang** [comment]
Standard function documentation:
- ST_GeomFromGeoJson: https://postgis.net/docs/ST_GeomFromGeoJSON.html
- ST_AsGeoJSON: https://postgis.net/docs/ST_AsGeoJSON.html

Seems `ST_GeogFromGeoJSON` is not a standard geo function?

File: pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java [hunk 1]
```diff
@@ -176,8 +176,13 @@ public enum TransformFunctionType {
   // Geo constructors
   ST_GEOG_FROM_TEXT("ST_GeogFromText", ReturnTypes.VARBINARY, OperandTypes.CHARACTER),
   ST_GEOM_FROM_TEXT("ST_GeomFromText", ReturnTypes.VARBINARY, OperandTypes.CHARACTER),
+
+  ST_GEOG_FROM_GEO_JSON("ST_GeogFromGeoJson", ReturnTypes.VARBINARY, OperandTypes.CHARACTER),
+  ST_GEOM_FROM_GEO_JSON("ST_GeomFromGeoJson", ReturnTypes.VARBINARY, OperandTypes.CHARACTER),
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/function/TransformFunctionType.java:181]: Keep the name consistent as the standard one, same for other places
```suggestion
  ST_GEOM_FROM_GEO_JSON("ST_GeomFromGeoJSON", ReturnTypes.VARBINARY, OperandTypes.CHARACTER),
```

File: pinot-core/src/main/java/org/apache/pinot/core/geospatial/transform/function/ConstructFromGeoJsonFunction.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import org.locationtech.jts.io.geojson.GeoJsonReader;
+
+
+/**
+ * An abstract class for implementing the geo constructor functions from GEO JSON.
+ */
+abstract class ConstructFromGeoJsonFunction extends BaseTransformFunction {
+
+  protected TransformFunction _transformFunction;
+  protected byte[][] _results;
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/geospatial/transform/function/ConstructFromGeoJsonFunction.java:44]: You can directly use `_bytesValuesSV`, and `initBytesValuesSV()` from the base class, same for other functions

File: pinot-core/src/main/java/org/apache/pinot/core/geospatial/transform/function/ConstructFromGeoJsonFunction.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    StringReader reader = new StringReader();
+
+    int length = valueBlock.getNumDocs();
+    for (int i = 0; i < length; i++) {
+      try {
+        reader.setString(argumentValues[i]);
+        Geometry geometry = _reader.read(reader);
+        _results[i] = GeometrySerializer.serialize(geometry);
+      } catch (ParseException e) {
+        Utils.rethrowException(
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/geospatial/transform/function/ConstructFromGeoJsonFunction.java:88]: No need to use re-throw for runtime exception

File: pinot-core/src/main/java/org/apache/pinot/core/geospatial/transform/function/StAsGeoJsonFunction.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+    try {
+      for (int i = 0; i < valueBlock.getNumDocs(); i++) {
+        Geometry geometry = GeometrySerializer.deserialize(values[i]);
+        GeometryUtils.GEO_JSON_WRITER.write(geometry, buffer);
+        _results[i] = buffer.getString();
+        buffer.clear();
+      }
+    } catch (IOException ioe) {
+      // should never happen
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/geospatial/transform/function/StAsGeoJsonFunction.java:87]: Even though it should not happen, let's wrap it as a runtime exception and throw it to guarantee it works as expected. Same for other places

File: pinot-core/src/main/java/org/apache/pinot/core/geospatial/transform/function/ConstructFromGeoJsonFunction.java (hunk 2 above)
Comment (bziobrowski) [reply pinot-core/src/main/java/org/apache/pinot/core/geospatial/transform/function/ConstructFromGeoJsonFunction.java]: I also fixed other functions in the package.

**Review by Jackie-Jiang — APPROVED** [review]

**Comment by Jackie-Jiang** [comment]
Great job! Please help also update the pinot documentation for these new functions

**Pull request merged**